MapboxCoreNavigation/RouteController.swift

```diff
@@ -330,6 +332,54 @@ extension RouteController: CLLocationManagerDelegate {
         }
     }
     
+    func checkForFasterRoute(from location: CLLocation) {
+        
+        // If the user is on a short route, don't check for faster alternatives
+        guard routeProgress.durationRemaining > 600 else { return }
+
+        // If the user is approaching a maneuver, don't check for a faster alternatives
+        guard routeProgress.currentLegProgress.currentStepProgress.durationRemaining > 70 else { return }
+        
+        guard let currentUpcomingManeuver = routeProgress.currentLegProgress.upComingStep else { return }
+        
+        guard let lastLocationTimestamp = lastLocationTimestamp else {
+            self.lastLocationTimestamp = location.timestamp
+            return
+        }
+
+        // Only check ever 2 minutes for faster route
+        guard location.timestamp.timeIntervalSince1970 - lastLocationTimestamp.timeIntervalSince1970 > 120 else { return }
+        
+        let options = routeProgress.route.routeOptions
+        options.waypoints = [Waypoint(coordinate: location.coordinate)] + routeProgress.remainingWaypoints
+        if let firstWaypoint = options.waypoints.first, location.course >= 0 {
+            firstWaypoint.heading = location.course
+            firstWaypoint.headingAccuracy = 90
+        }
+        
+        let durationRemaining = routeProgress.durationRemaining
+        let currentAlertLevel = routeProgress.currentLegProgress.alertUserLevel
+        
+        routeTask?.cancel()
+        routeTask = directions.calculate(options, completionHandler: { [weak self] (waypoints, routes, error) in
+            guard let strongSelf = self else { return }
+            
+            strongSelf.lastLocationTimestamp = nil
+            
+            if let route = routes?.first {
+                let percentDifference = ((durationRemaining - route.expectedTravelTime) / route.expectedTravelTime) * 100
```

Comment:
If the user is currently 10 minutes from their destination and the new route is 9 minutes long, that’s a 10% improvement, but `percentDifference` is 11.1 because it puts the new route’s time in the denominator instead of the old route’s time.